src/client/app/containers/BarChartContainer.ts

```diff
@@ -97,16 +103,26 @@ function mapStateToProps(state) {
 			backgroundColor: 'rgba(0,0,0,0.6)',
 			displayColors: false,
 			callbacks: {
-				label: tooltipItems => `${data.datasets[tooltipItems.datasetIndex].label}: ${tooltipItems.yLabel} kW`
+				label: (tooltipItems: ChartTooltipItem) => {
+					if (tooltipItems.datasetIndex !== undefined) {
+						return `${datasets[tooltipItems.datasetIndex].label}: ${tooltipItems.yLabel} kW`;
+					} else {
+						throw new Error('tooltipItems.datasetIndex was undefined in line chart tooltip label callback');
+					}
+				}
 			}
 		}
 	};
 
-	return {
+	const data: ChartData = { datasets, labels };
+
+	const props: ChartComponentProps =  {
```

Comment:
This seems redundant. Is there a way to not type something that immediately gets returned?

Reply:
The type errors are much nicer this way.